### PR TITLE
Fix Bug on Special Dialog

### PR DESCRIPTION
**Pull request by leleOFA**
## Short description
fix initial value of add special contant dialog

## List of changes proposed in this pull request
-delete an wrong intiali value setted

## How to test
on the recapiti page click on personalize contact

NB a pair of test skipped for moment

### CODE REVIEW

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 1]
```diff
@@ -61,10 +61,7 @@ describe('test AddSpecialContactDialog', () => {
     expect(titleEl).toHaveTextContent(`special-contacts.modal-title`);
     const bodyEl = within(dialog).getByTestId('dialog-content');
     expect(bodyEl).toBeInTheDocument();
-    expect(bodyEl).toHaveTextContent(`special-contacts.pec`);
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:64]: Why this change?

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ const AddSpecialContactDialog: React.FC<Props> = ({
     checkIfSenderIsAlreadyAdded(formik.values.sender, e.target.value as ChannelType);
   };
 
-  const checkIfSenderIsAlreadyAdded = (sender: Party, channelType: ChannelType) => {
+  const checkIfSenderIsAlreadyAdded = (sender: Party, channelType: ChannelType | string) => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:65]: Remove the | string and put an if check in addressTypeChangeHandler and senderChangeHandler

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 3]
```diff
@@ -117,27 +117,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
       }),
   });
 
-  const initialValues = {
+  const initialValues: {
+    sender: { id: string; name: string | '' };
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:121]: remove the | ''

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 4]
```diff
@@ -117,27 +117,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-    channelType: value
-      ? channelType
-      : addressTypes.find((a) => !a.disabled && a.shown)?.id ?? ChannelType.PEC,
+    channelType: value ? channelType : '',
     s_value: value,
   };
 
   const formik = useFormik({
     initialValues,
+    initialErrors: { ...initialValues },
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:135]: why this?

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 5]
```diff
@@ -217,22 +222,23 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-              })}
-              value={formik.values.s_value}
-              onChange={handleChangeTouched}
-              error={formik.touched.s_value && Boolean(formik.errors.s_value)}
-              helperText={formik.touched.s_value && formik.errors.s_value}
-              sx={{ mb: 2 }}
-            />
-          )}
+          {formik.values.channelType !== ChannelType.SERCQ_SEND &&
+            formik.values.channelType !== '' && (
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:226]: change the formik.values.channelType !== ChannelType.SERCQ_SEND && formik.values.channelType !== '' with formik.values.channelType === ChannelType.PEC

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 6]
```diff
@@ -61,10 +61,7 @@ describe('test AddSpecialContactDialog', () => {
     expect(titleEl).toHaveTextContent(`special-contacts.modal-title`);
     const bodyEl = within(dialog).getByTestId('dialog-content');
     expect(bodyEl).toBeInTheDocument();
-    expect(bodyEl).toHaveTextContent(`special-contacts.pec`);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:64]: Don't remove the check on input. Replace getById with queryById and check that the input is not shown

File: packages/pn-personafisica-webapp/src/utility/contacts.utility.ts [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { SelectedAddresses } from '../redux/contact/reducers';
 
 export const internationalPhonePrefix = '+39';
 
-type AddressTypeItem = {
+export type AddressTypeItem = {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/utility/contacts.utility.ts:11]: remove this export

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 8]
```diff
@@ -62,7 +62,7 @@ const AddSpecialContactDialog: React.FC<Props> = ({
     checkIfSenderIsAlreadyAdded(formik.values.sender, e.target.value as ChannelType);
   };
 
-  const checkIfSenderIsAlreadyAdded = (sender: Party, channelType: ChannelType) => {
+  const checkIfSenderIsAlreadyAdded = (sender: Party, channelType: ChannelType | '') => {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:65]: remove the | '' and put an if check in addressTypeChangeHandler and senderChangeHandler

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 9]
```diff
@@ -117,27 +117,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
       }),
   });
 
-  const initialValues = {
+  const initialValues: {
+    sender: { id: string; name: string | '' };
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:121]: remove the | ''

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 10]
```diff
@@ -117,27 +117,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-    channelType: value
-      ? channelType
-      : addressTypes.find((a) => !a.disabled && a.shown)?.id ?? ChannelType.PEC,
+    channelType: value ? channelType : '',
     s_value: value,
   };
 
   const formik = useFormik({
     initialValues,
+    initialErrors: { ...initialValues },
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:135]: why this?

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 11]
```diff
@@ -217,22 +222,23 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-                ns: 'recapiti',
-              })}
-              value={formik.values.s_value}
-              onChange={handleChangeTouched}
-              error={formik.touched.s_value && Boolean(formik.errors.s_value)}
-              helperText={formik.touched.s_value && formik.errors.s_value}
-              sx={{ mb: 2 }}
-            />
-          )}
+          {formik.values.channelType !== ChannelType.SERCQ_SEND &&
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:225]: change the formik.values.channelType !== ChannelType.SERCQ_SEND && formik.values.channelType !== '' with formik.values.channelType === ChannelType.PEC

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx (hunk 1 above)
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:64]: Don't remove the check on input. Replace getById with queryById and check that the input is not shown

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 2 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:65]: doone

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 3 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:121]: dne

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 4 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:135]: for set bug on button. The buttoon is active without value popolate. the paramenter in submit 'isValid' check on errors array in formik, this is usefull for popoolate this array

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 5 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:226]: dopne

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx (hunk 6 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:64]: donoe

File: packages/pn-personafisica-webapp/src/utility/contacts.utility.ts (hunk 7 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/utility/contacts.utility.ts:11]: done

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 8 above)
Comment (leleOFA) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:65]: done

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 9 above)
Comment (leleOFA) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:121]: done

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx (hunk 1 above)
Comment (leleOFA) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:64]: doone

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 12]
```diff
@@ -61,10 +61,8 @@ describe('test AddSpecialContactDialog', () => {
...
-    const input = getById(bodyEl, 's_value');
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:65]: restore this code and use queryById and check that the input is not in the document

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 13]
```diff
@@ -117,27 +123,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
+    initialErrors: { ...initialValues },
     validateOnMount: true,
     validationSchema,
     enableReinitialize: true,
     onSubmit: (values) => {
-      onConfirm(values.s_value, values.channelType, {
-        senderId: values.sender.id,
-        senderName: values.sender.name,
-      });
+      if (values.channelType !== '') {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:146]: Do the same change requested on pf

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 14]
```diff
@@ -61,10 +61,8 @@ describe('test AddSpecialContactDialog', () => {
...
-    const input = getById(bodyEl, 's_value');
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:65]: Do the same changes requested on pf

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 4 above)
Comment (AndreaCimini90) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:142]: this isn't the correct way because the initialErrors should not be populated with the initialValues object. Furthermore the bug seems to be not present

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 10 above)
Comment (AndreaCimini90) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:141]: same as for pf

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 15]
```diff
@@ -58,8 +58,10 @@ const AddSpecialContactDialog: React.FC<Props> = ({
 
   const addressTypeChangeHandler = async (e: ChangeEvent<HTMLInputElement>) => {
     await formik.setFieldValue('s_value', '');
-    formik.handleChange(e);
-    checkIfSenderIsAlreadyAdded(formik.values.sender, e.target.value as ChannelType);
+    if (e.target.value !== '') {
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:67]: done

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 16]
```diff
@@ -81,7 +83,12 @@ const AddSpecialContactDialog: React.FC<Props> = ({
   const senderChangeHandler = async (_: any, newValue: Party | null) => {
     await formik.setFieldTouched('sender', true, false);
     await formik.setFieldValue('sender', { id: newValue?.id ?? '', name: newValue?.name ?? '' });
-    if (newValue && addressesData.addresses.some((a) => a.senderId === newValue.id)) {
+    if (
+      newValue &&
+      addressesData.addresses.some(
+        (a) => a.senderId === newValue.id && formik.values.channelType !== ''
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:104]: done

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 17]
```diff
@@ -117,27 +124,32 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
+    initialErrors: { ...initialValues },
     validateOnMount: true,
     validationSchema,
     enableReinitialize: true,
     onSubmit: (values) => {
-      onConfirm(values.s_value, values.channelType, {
-        senderId: values.sender.id,
-        senderName: values.sender.name,
-      });
+      if (values.channelType !== '') {
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:165]: doone

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 18]
```diff
@@ -370,7 +385,7 @@ describe('test AddSpecialContactDialog', () => {
       true
     );
 
-    expect(confirmButton).toBeDisabled();
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:373]: Why did you change this check?

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 19]
```diff
@@ -97,18 +99,7 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-        pecInValidationForSender(newValue?.id ?? '')
-      )
-    ) {
-      setIsPecInValidation(false);
-    } else {
-      setIsPecInValidation(true);
-    }
-
-    if (newValue && addressesData.addresses.some((a) => a.senderId === newValue.id)) {
+    if (newValue && addressesData.addresses.some((a) => a.senderId === newValue.id) && formik.values.channelType) {
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:102]: Here you missed a code from develop that mustn't be removed

File: packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx [hunk 20]
```diff
@@ -370,7 +381,7 @@ describe('test AddSpecialContactDialog', () => {
       true
     );
 
-    expect(confirmButton).toBeDisabled();
+    expect(confirmButton).toBeEnabled();
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:384]: Same as pf

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx [hunk 21]
```diff
@@ -97,23 +99,13 @@ const AddSpecialContactDialog: React.FC<Props> = ({
...
-        pecInValidationForSender(newValue?.id ?? '')
-      )
-    ) {
-      setIsPecInValidation(false);
-    } else {
-      setIsPecInValidation(true);
-    }
-
-    if (newValue && addressesData.addresses.some((a) => a.senderId === newValue.id)) {
+    if (newValue && addressesData.addresses.some((a) => a.senderId === newValue.id) && formik.values.channelType) {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:102]: Here you missed a code from develop that mustn't be removed

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 21 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx]: done

File: packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx (hunk 19 above)
Comment (leleOFA) [reply packages/pn-personagiuridica-webapp/src/components/Contacts/AddSpecialContactDialog.tsx:114]: done

File: packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx (hunk 18 above)
Comment (leleOFA) [reply packages/pn-personafisica-webapp/src/components/Contacts/__test__/AddSpecialContactDialog.test.tsx:373]: the button result enabled
